include/dev/usb.h

```diff
@@ -145,4 +149,21 @@ typedef struct usb_endpoint_descriptor {
 #define UE_BULK 0x02
 #define UE_INTERRUPT 0x03
 
+/*
+ * Implementation specific constructs.
+ */
+
+typedef enum {
+  UEF_STALLED = 1, /* STALL condition encountered */
+  UEF_ERROR = 2,   /* errors other than STALL */
+} usb_error_flags_t;
+
+/* USB buffer used for USB transfers. */
+typedef struct usb_buf {
+  ringbuf_t buf;           /* write source or read destination */
+  condvar_t cv;            /* wait for the transfer to complete */
+  spin_t lock;             /* buffer guard */
+  usb_error_flags_t flags; /* errors encountered during transfer */
```

Comment:
`eflags` or `errors` would be better imo, but this is fine as well , especially if there's a possiblity that this field may get expanded to include more information than wether particular errors were encountered.

Reply:
There are some other flags now and thus I have applied your suggestion.